### PR TITLE
Add htmlspecialchars when showing $DATE_AT

### LINKED ISSUE #2061: [Security] Refected XSS in doku.php
## Bug
Field `at` is not sanitized in `msg` error message. It's a reflected XSS.

## Detail
`doku.php`
```
$DATE_AT = $INPUT->str('at');
...
msg(sprintf($lang['unable_to_parse_date'], $DATE_AT));
```

## PoC
`http://localhost/dokuwiki/doku.php?id=wiki:welcome&at=<svg onload=alert(cookie)>`

### PR DESCRIPTION
**Pull request by phy25**
Fixes #2061.

By the way, maybe a thorough check with all `msg()` usage is necessary.

### CODE REVIEW

File: doku.php [hunk 1]
```diff
@@ -62,7 +62,7 @@
     } else { // check for UNIX Timestamp
         $date_parse = @date('Ymd',$DATE_AT);
         if(!$date_parse || $date_parse === '19700101') {
-            msg(sprintf($lang['unable_to_parse_date'], $DATE_AT));
+            msg(sprintf($lang['unable_to_parse_date'], htmlspecialchars($DATE_AT)));
```
Comment (splitbrain) [inline doku.php]: This should use our own shortcut `hsc()` instead.

Comment (phy25) [inline doku.php]: That's good. I got the idea from `inc/actions.php`. Will fix that in this pull request as well.

**Comment by phy25** [comment]
Stuffs that are not replaced with `hsc` yet:
- Plugins bundled with core
- Libraries (expect `msg()` is used)

File: install.php [hunk 2]
```diff
@@ -161,7 +161,7 @@ function print_form($d){
     include(DOKU_CONF.'license.php');
 
     if(!is_array($d)) $d = array();
-    $d = array_map('htmlspecialchars',$d);
+    $d = array_map('hsc',$d);
```
Comment (splitbrain) [inline install.php]: Did you check if hsc is available in install.php already? It used to be that install.php had not the full environment initialized, yet.

Comment (splitbrain) [inline install.php]: okay. seems to be available.

**Pull request merged**